app/controllers/AtsMergePageController.scala

```diff
@@ -58,15 +58,20 @@ class AtsMergePageController @Inject()(
 
     atsMergePageService.getSaAndPayeYearList.map {
       case Right(atsMergePageViewModel) =>
-        Ok(
-          atsMergePageView(
-            atsMergePageViewModel,
-            formWithErrors.getOrElse(atsForms.atsYearFormMapping),
+        val actingAsAttorneyFor: Option[ActingAsAttorneyFor] = atsMergePageViewModel.saData.utr.isEmpty match {
+          case true => None
+          case false =>
             getActingAsAttorneyFor(
               request,
               atsMergePageViewModel.saData.forename,
               atsMergePageViewModel.saData.surname,
               atsMergePageViewModel.saData.utr)
```

Comment:
Add your check for utr in getActingAsAttorneyFor. You already return an option so it would be more logical to have the check in there.